cli/azd/pkg/environment/azdcontext/azdcontext.go

```diff
@@ -147,15 +147,29 @@ func (c *AzdContext) NewEnvironment(name string) error {
 	return nil
 }
 
+// Creates context with project directory set to the desired directory.
+func NewAzdContextWithDirectory(projectDirectory string) *AzdContext {
+	return &AzdContext{
+		projectDirectory: projectDirectory,
+	}
+}
+
+var (
+	errNoProject = errors.New("no project exists; to create a new project, run `azd init`")
+)
+
+// Creates context with project directory set to the nearest project file found.
+//
+// The project file is first searched for in the current directory, if not found, the parent directory is searched
+// recursively up to root. If no project file is found, errNoProject is returned.
```

Comment:
We should either export `errNoProject` or update this comment to say something like "If no project file is found, an error is returned).

At one point I had made this a named error because I thought people might want to test against it, but it looks like we never really used that ability. I'm happy if we don't want to export it for now.